src/Microsoft.Health.Fhir.Liquid.Converter.UnitTests/Filters/DateFiltersTests.cs

```diff
@@ -15,12 +15,45 @@ public class DateFiltersTests
     {
         public static IEnumerable<object[]> GetValidDataForAddHyphensDate()
         {
-            yield return new object[] { null, null };
-            yield return new object[] { string.Empty, string.Empty };
-            yield return new object[] { @"2001", @"2001" };
-            yield return new object[] { @"200101", @"2001-01" };
-            yield return new object[] { @"19241010", @"1924-10-10" };
-            yield return new object[] { @"19850101000000", @"1985-01-01" };
+            yield return new object[] { null, "local", null };
+            yield return new object[] { string.Empty, "local", string.Empty };
+            yield return new object[] { @"2001", "preserve", @"2001" };
+            yield return new object[] { @"200101", "preserve", @"2001-01" };
+            yield return new object[] { @"19241010", "local", @"1924-10-10" };
+            yield return new object[] { @"19850101000000", "local", @"1985-01-01" };
+        }
+
+        // We assume the local timezone is +08:00.
+        public static IEnumerable<object[]> GetValidDataWithoutTimeZoneForAddHyphensDateWithUtcTimeZoneHandling()
+        {
+            yield return new object[] { @"200101", "utc", @"2000-12" };
+            yield return new object[] { @"20010102", "utc", @"2001-01-01" };
+            yield return new object[] { @"19850101000000", "utc", @"1984-12-31" };
+        }
+
+        public static IEnumerable<object[]> GetValidDataForAddSeconds()
```

Comment:
Can we add some double seconds instead of integer seconds here (e.g., 60.5)?

Reply:
Added here, thanks.